File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/UpdateLayerExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class UpdateLayerExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var layerPatchInput = new LayerPatchInput(
            instructions: JsonSerializer.Deserialize<List<Dictionary<string, object>>>("""
                [
                    {
                        "experimentKey": "checkout-button-color",
                        "kind": "updateExperimentReservation",
                        "reservationPercent": 25
                    }
                ]
            """),
            comment: "Example comment describing the update",
            environmentKey: "production"
        );

        try
        {
            var response = new LayersApi(config).UpdateLayer(
                projectKey: null,
                layerKey: null,
                layerPatchInput: layerPatchInput
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling LayersApi#UpdateLayer: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/UpdateLayerExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const layerPatchInput: api.LayerPatchInput = {
  instructions: [
    {
      "experimentKey": "checkout-button-color",
      "kind": "updateExperimentReservation",
      "reservationPercent": 25
    }
  ],
  comment: "Example comment describing the update",
  environmentKey: "production",
};

new api.LayersApi(configuration).updateLayer(
  "projectKey_string", // projectKey
  "layerKey_string", // layerKey
  layerPatchInput,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling LayersApi#updateLayer:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
